tapgarden/planter.go

```diff
@@ -141,6 +144,66 @@ type PendingAssetGroup struct {
 	asset.GroupVirtualTx
 }
 
+// PSBT returns a PSBT packet that can be used to create a group witness for the
+// asset group.
+func (p *PendingAssetGroup) PSBT(
+	params chaincfg.Params) (*psbt.Packet, error) {
+
+	// Generate PSBT equivalent of the group virtual tx.
+	packet, err := psbt.NewFromUnsignedTx(&p.GroupVirtualTx.Tx)
+	if err != nil {
+		return nil, fmt.Errorf("error producing group virtual PSBT "+
+			"from tx: %w", err)
+	}
+
```

Comment:
Re my other comment, this is where we'd add the new custom global fields to allow external signers/coordinates to display meta data w.r.t what it actually being minted. We can also provide the information needed to derive the virtual tx in the first place (for additional validation).

Reply:
Great idea!

I've spun this into a new issue: https://github.com/lightninglabs/taproot-assets/issues/1298
I did that because I would like to shape it a little and I don't think it's critical right now. But would be good to get it into the same release.